core/src/main/java/io/grpc/internal/SerializingExecutor.java

```diff
@@ -135,39 +135,31 @@ public void execute(Runnable r) {
   private class TaskRunner implements Runnable {
     @Override
     public void run() {
-      boolean stillRunning = true;
-      try {
-        while (true) {
-          Runnable nextToRun;
+      Runnable nextToRun;
+      ArrayDeque<Runnable> tmp = null;
+      do {
+        if (tmp == null) {
```

Comment:
Switched back to that, also switched back to allowing Error to pass through the catch